src/raw/cgroup.go

```diff
@@ -6,20 +6,26 @@ import (
 	"os"
 	"strconv"
 	"strings"
+
+	"github.com/docker/docker/api/types"
+)
+
+const (
+	CgroupV2 = "2"
 )
 
-// NewCgroupsFetcher creates the proper metrics fetcher for the used cgroups version.
-func NewCgroupsFetcher(
-	hostRoot string,
-	cgroupInfo *CgroupInfo,
-	systemCPUReader SystemCPUReader,
-	networkStatsGetter NetworkStatsGetter,
-) (Fetcher, error) {
-	if cgroupInfo.Version == CgroupV2 {
-		return NewCgroupsV2Fetcher(hostRoot, cgroupInfo.Driver, NewCgroupV2PathParser(), systemCPUReader, networkStatsGetter)
+// NewCgroupFetcher returns either a V2Fetcher or a V1Fetcher depending on the cgroupInfo
+func NewCgroupFetcher(hostRoot string, cgroupInfo types.Info) (Fetcher, error) {
+	detectedHostRoot, err := DetectHostRoot(hostRoot, CanAccessDir)
+	if err != nil {
+		return nil, err
 	}
 
-	return NewCgroupsV1Fetcher(hostRoot, NewCgroupV1PathParser(), systemCPUReader, networkStatsGetter)
+	if cgroupInfo.CgroupVersion == CgroupV2 {
+		return NewCgroupsV2Fetcher(detectedHostRoot, cgroupInfo.CgroupDriver, NewPosixSystemCPUReader())
+	}
+	// if cgroupInfo.Version == CgroupV1
```

Comment:
looks like there is a reason to pick v1 by default , do you know if there is any corner-case when the API doesn't provide this info and is v1?

I guess this line is used in the following PR but just in case I saw the commented code.

Reply:
I think that it is there for legacy reasons. We could error out since it should be impossible. But since we are not 100% sure we have that behavior